code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -1468,3 +1468,96 @@
 		for(var/X in actions)
 			var/datum/action/act = X
 			act.update_button_icon()
+
+// Crusher Horn Toss
+/mob/living/carbon/Xenomorph/proc/cresttoss(var/mob/living/carbon/M)
+	if (cresttoss_used)
+		return
+
+	if (!check_plasma(40))
+		return
+
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to fling away [M] but are unable as you fail to shake off the shock!</span>")
+		return
+
+	if (!Adjacent(M)) //Sanity check
+		return
+
+	if(!istype(M, /mob/living)) //Sanity check
+		return
+
+	if(M.mob_size >= MOB_SIZE_BIG) //We can't fling big aliens/mobs
+		to_chat(src, "<span class='xenowarning'>[M] is too large to fling!</span>")
+		return
+
+	icon_state = "Crusher Charging"  //Momentarily lower the crest for visual effect
+	visible_message("<span class='xenowarning'>\The [src] flings [M] away with its crest!</span>", \
+	"<span class='xenowarning'>You fling [M] away with your crest!</span>")
+
+	cresttoss_used = 1
+	use_plasma(40)
+
+	face_atom(M) //Face towards the target so we don't look silly
+
+	if(M.stat != DEAD && (!(M.status_flags & XENO_HOST) || !istype(M.buckled, /obj/structure/bed/nest)) )
+		var/facing = get_dir(src, M)
+		var/toss_distance = rand(3,5)
+		var/turf/T = loc
+		var/turf/temp = loc
+		if(a_intent == "hurt") //If we use the ability on hurt intent, we throw them in front; otherwise we throw them behind.
+			for (var/x = 0, x < toss_distance, x++)
+				temp = get_step(T, facing)
+				if (!temp)
+					break
+				T = temp
+		else
+			facing = get_dir(M, src)
+			M.loc = get_step(T, facing) //Move the target behind us before flinging
+			for (var/x = 0, x < toss_distance, x++)
+				temp = get_step(T, facing)
+				if (!temp)
+					break
+				T = temp
+		//The target location deviates up to 1 tile in any direction
+		var/scatter_x = rand(-1,1)
+		var/scatter_y = rand(-1,1)
+		var/turf/new_target = locate(T.x + round(scatter_x),T.y + round(scatter_y),T.z) //Locate an adjacent turf.
+		if(new_target)
+			T = new_target//Looks like we found a turf.
+
+		M.throw_at(T, toss_distance, 1, src)
+		M.KnockDown(1, 1)
+
+		//Handle the damage
+		if(!isXeno(M)) //Friendly xenos don't take damage.
+			var/damage = toss_distance * 5
+			if(frenzy_aura > 0)
+				damage *= (1 + round(frenzy_aura * 0.1,0.01))
+			var/armor_block = M.run_armor_check("chest", "melee")
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.take_overall_damage(rand(damage * 0.75,damage * 1.25) * 0.5, armor_block) //Armour functions against this.
```

Comment:
Redundant typecheck

Reply:
As above, these procs work differently for non-human mobs vs humans.